src/main/java/com/scalar/db/storage/cosmos/CosmosOperation.java

```diff
@@ -0,0 +1,107 @@
+package com.scalar.db.storage.cosmos;
+
+import com.azure.cosmos.models.PartitionKey;
+import com.google.common.base.Joiner;
+import com.scalar.db.api.Operation;
+import com.scalar.db.io.Value;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+
+/** A class to treating utilities for a operation */
+public class CosmosOperation {
+  private final Operation operation;
+  private final TableMetadata metadata;
+
+  public CosmosOperation(Operation operation, TableMetadata metadata) {
```

Comment:
Since it seems always getting `TableMetadata` for the specified operation,
it might be better to pass`MetadataManager` object so that we can simplify the code.

```suggestion
  public CosmosOperation(Operation operation, TableMetadataManager metadataManager) {
```

Reply:
Thank you! I've fixed that.